modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AspNetCoreServerCodegen.java

```diff
@@ -342,15 +381,23 @@ public String getNullableType(Schema p, String type) {
 
     private void setCliOption(CliOption cliOption) throws IllegalArgumentException {
         if (additionalProperties.containsKey(cliOption.getOpt())) {
-            cliOption.setOptValue(additionalProperties.get(cliOption.getOpt()).toString());
-            if (cliOption.getOptValue() == null) {
-                cliOption.setOptValue(cliOption.getDefault());
-                throw new IllegalArgumentException(cliOption.getOpt() + ": Invalid value '" + additionalProperties.get(cliOption.getOpt()).toString() + "'" +
-                        ". " + cliOption.getDescription());
+            // TODO Hack - not sure why the empty strings become boolean.
+            Object obj = additionalProperties.get(cliOption.getOpt());
```

Comment:
I'm not sure how I feel about a generator setting a CLI option. There shouldn't be a need to sync options to CLI options, because it means those options can change after the point that they're expected to be read and used to configure the generator. 

What logical flow did you find a need to keep these on sync? Ideally, only the CLI should be there owner of seeing these values (to remove confusion described above).

Reply:
Its really a hack - the generator seems to change empty strings to the boolean false -its actually to fix that. I will open an issue to see why that happens but fixing it will require care in case other generators depend on this conversion.